splunk/centos-7/Dockerfile

```diff
@@ -11,61 +11,64 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-FROM base-centos-7:latest
-LABEL maintainer="support@splunk.com"
 
+#
+# Download and unpack Splunk Enterprise
+#
+FROM base-debian-9:latest as package
 ARG SPLUNK_FILENAME
 ARG SPLUNK_BUILD_URL
+ENV SPLUNK_HOME=/opt/splunk
+RUN echo "Downloading Splunk and validating the checksum at: ${SPLUNK_BUILD_URL}"
+RUN wget -qO /tmp/${SPLUNK_FILENAME} ${SPLUNK_BUILD_URL}
+RUN wget -qO /tmp/${SPLUNK_FILENAME}.sha512 ${SPLUNK_BUILD_URL}.sha512
+RUN (cd /tmp && sha512sum -c ${SPLUNK_FILENAME}.sha512)
+RUN tar -C /opt -zxf /tmp/${SPLUNK_FILENAME}
+RUN mv ${SPLUNK_HOME}/etc ${SPLUNK_HOME}-etc
+RUN mkdir -p ${SPLUNK_HOME}/etc ${SPLUNK_HOME}/var
+
+
+
+FROM base-centos-7:latest
+LABEL maintainer="support@splunk.com"
+
 ARG SPLUNK_DEFAULTS_URL
 
 ENV SPLUNK_HOME=/opt/splunk \
     SPLUNK_GROUP=splunk \
     SPLUNK_USER=splunk \
     SPLUNK_ROLE=splunk_standalone \
-    SPLUNK_FILENAME=${SPLUNK_FILENAME} \
     SPLUNK_DEFAULTS_URL=${SPLUNK_DEFAULTS_URL} \
     SPLUNK_ANSIBLE_HOME=/opt/ansible \
     ANSIBLE_USER=ansible \
     ANSIBLE_GROUP=ansible \
     CONTAINER_ARTIFACT_DIR=/opt/container_artifact
 
-# Setup users and download Splunk
+# Copy ansible playbooks
+COPY splunk-ansible ${SPLUNK_ANSIBLE_HOME}
```

Comment:
In terms of ordering, I think I would prefer this COPY step + the one on line 51 to be after the RUN command that sets up users/directories? 

My reasoning is during development, we're more likely to update `splunk-ansible` frequently. Putting the COPY step here would force all the subsequent steps to rebuild when it's probably doesn't necessarily need to. Granted, everything in the RUN step should be quicker now that we've removed the "download Splunk" bit, but just a matter of quality of life.

Maybe ordering should be something the following?
* RUN groupadd ...
* COPY --from=package ...
* COPY [ "splunk/common-files/updateetc.sh" ...
* COPY splunk-ansible ...

Reply:
The problem I had is that some of the RUN commands assume that the copies have taken place (chmods and such). I thought it was better for readability to just keep the COPYs together and consolidate the RUNs, and it also reduces the number of layers produced vs splitting up the RUN commands.

Reply:
And like you said, the RUN step is relatively fast now so I don't think this ordering impacts dev cycle time anymore.